File: pinyougouParent/pinyougou-user-web/src/main/java/com/pinyougou/user/controller/UserController.java
package com.pinyougou.user.controller;

import com.pinyougou.pojo.Result;
import com.pinyougou.pojo.TbUser;
import com.pinyougou.user.service.UserService;
import com.alibaba.dubbo.config.annotation.Reference;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;
import util.PhoneFormatCheckUtils;

@RestController
@RequestMapping("/user")
public class UserController {

   @Reference
   private UserService userService;

    @RequestMapping("/sendCode")
    public Result sendCode(String phone){
        //判断手机号格式
        if(!PhoneFormatCheckUtils.isPhoneLegal(phone)){
            return new Result(false,"手机号格式不正确");
        }
        try {
            userService.createSmsCode(phone);
            System.out.println(userService);
            return new Result(true,"验证码发送成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false,"验证码发送失败");

        }
    }

    @RequestMapping("/add")
    public Result add(@RequestBody TbUser user,String smscode){
        boolean b = userService.checkSmsCode(user.getPhone(), smscode);
        if(b==false){
            return new Result(false,"验证码输入错误！");
        }
        try {
            System.out.println(userService);
            userService.add(user);
            return new Result(true,"增加成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false,"增加失败");
        }
    }
}

File: pinyougouParent/pinyougou-user-web/src/main/webapp/js/controller/userController.js
 //控制层 
app.controller('userController' ,function($scope,$controller   ,userService){	
	
	//注册用户
	$scope.reg=function(){
		if($scope.entity.password!=$scope.password){
			alert("两次密码输入不一致，请重新输入");
			return;
		}
		userService.add($scope.entity,$scope.smscode).success(
			function(response){
				alert(response.message);
			}
		)
	}
    
	//发送验证码
	$scope.sendCode=function(){
		if($scope.entity.phone==null){
			alert("请输入手机号！");
			return;
		}
		userService.sendCode($scope.entity.phone).success(
			function(response){
				alert(response.message)
			}
		)
	}
	
});	
